pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -33,7 +33,15 @@ cmakeConfigurePhase() {
         # By now it supports linux builds only. We should set the proper
         # CMAKE_SYSTEM_NAME otherwise.
         # http://www.cmake.org/Wiki/CMake_Cross_Compiling
-        cmakeFlags="-DCMAKE_CXX_COMPILER=$crossConfig-g++ -DCMAKE_C_COMPILER=$crossConfig-gcc $cmakeFlags"
+        #
```

Comment:
Could we add `-DCMAKE_CROSSCOMPILING` here by any chance?

Reply:
I could be mistaken or remembering incorrectly but... I don't know that that is something one normally sets explicitly.  Rather, we set it when building LLVM so things works properly despite us not properly leveraging cmake's cross compilation features (which normally would set that value for us, maybe?).

The "right" way involves... I honestly don't know O:).  What we do here thought doesn't seem obviously broken in general, with other cmake-based builds, so maybe we're only missing something minor.

FWIW: https://cmake.org/Wiki/CMake_Cross_Compiling

Reply:
Ha! I was googling/grep'ing using the wrong string, silly rabbit.

Yes, I remembered correctly.  Here's the docs for that variable: https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html

(which imply if cmake "knew" about the cross-compilation we were doing it would set this to true for us)